src/SprykerEco/Zed/NewRelic/Business/Model/RecordDeployment.php

```diff
@@ -66,23 +81,51 @@ public function recordDeployment(array $arguments = []): RecordDeploymentInterfa
      */
     protected function createRecordDeploymentRequest(array $params): ResponseInterface
     {
+        $applicationId = $params['application_id'] ?? null;
+
+        unset($params['app_name']);
+        unset($params['application_id']);
+
         $options = [
             'headers' => [
-                'x-api-key' => $this->newRelicApiKey,
+                'X-Api-Key' => $this->newRelicApiKey,
+            ],
+            'json' => [
+                'deployment' => $params,
             ],
         ];
 
-        $data = [];
-        $data['deployment'] = [];
-        foreach ($params as $key => $value) {
-            $data['deployment'][$key] = $value;
-        }
-        $options['form_params'] = $data;
-
         $httpClient = new Client();
 
-        $request = $httpClient->post($this->newRelicDeploymentApiUrl, $options);
+        $deploymentUrl = $this->newRelicDeploymentApiUrl;
+        if ($applicationId) {
+            $deploymentUrl = sprintf($this->newRelicDeploymentApiUrl, $applicationId);
+        }
+
+        $request = $httpClient->post($deploymentUrl, $options);
 
         return $request;
     }
+
+    /**
+     * @param array $arguments
+     *
+     * @throws \SprykerEco\Zed\NewRelic\Business\Exception\RecordDeploymentException
+     *
+     * @return $this
+     */
+    private function recordSingleDeployment(array $arguments=[]): RecordDeploymentInterface
+    {
+        $response = $this->createRecordDeploymentRequest($arguments);
+        $statusCode = $response->getStatusCode();
+        if ($statusCode < static::STATUS_CODE_SUCCESS || $statusCode >= static::STATUS_CODE_REDIRECTION) {
+            throw new RecordDeploymentException(sprintf(
```

Comment:
Will that break the deployment? Shouldn't we only log here?

Reply:
This is the exactly same throw exception of here: https://github.com/spryker-eco/new-relic/pull/3/files#diff-58b5cba6df57adf679f01002327b7cb6L52
But, this might also run in a loop of IDs. I think it's the right behaviour anyway. Even with several applications, deployment shouldn't be an issue (it would be only if one of IDs is wrong, I guess).